trestle/tasks/xlsx_to_oscal_profile.py

```diff
@@ -123,20 +126,90 @@ def _execute(self) -> TaskOutcome:
+            include_controls=[SelectControlById(with_ids=self._get_with_ids_by_rule())],
+        )
+        imports = [import_]
+        return imports
+
+    def _get_imports_by_check(self) -> List[Import]:
+        """Get imports by check."""
+        import_ = Import(
+            href=self._get_spread_sheet_url(),
+            include_controls=[SelectControlById(with_ids=self._get_with_ids_by_check())],
+        )
+        imports = [import_]
+        return imports
+
+    def _get_with_ids_by_control(self) -> List[str]:
+        """Get controls from spread sheet."""
+        control_list = []
+        for row in self.xlsx_helper.row_generator():
+            # quit when first row with no goal_id encountered
+            controls = self.xlsx_helper.get_controls(row)
+            if controls is not None:
+                for control in controls:
+                    control = self._oscal_namify(control)
+                    if control in control_list:
+                        continue
+                    control_list.append(control)
+        return sorted(control_list, key=self._control_sort_key)
+
+    def _get_with_ids_by_rule(self) -> List[str]:
+        """Get rules from spread sheet."""
+        rule_name_id_list = []
+        for row in self.xlsx_helper.row_generator():
+            # quit when first row with no goal_id encountered
+            rule_name_id = self.xlsx_helper.get_rule_name_id(row, strict=True)
+            if rule_name_id is not None:
+                if rule_name_id in rule_name_id_list:
+                    continue
+                rule_name_id_list.append(rule_name_id)
+        return sorted(rule_name_id_list)
+
+    def _get_with_ids_by_check(self) -> List[str]:
+        """Get check from spread sheet."""
+        check_name_id_list = []
         for row in self.xlsx_helper.row_generator():
             # quit when first row with no goal_id encountered
-            goal_name_id = self.xlsx_helper.get_goal_name_id_strict(row)
-            if goal_name_id is not None:
-                goal_name_id_list.append(goal_name_id)
-        return sorted(goal_name_id_list)
+            check_name_id = self.xlsx_helper.get_check_name_id(row, strict=True)
+            if check_name_id is not None:
+                if check_name_id in check_name_id_list:
+                    continue
+                check_name_id_list.append(check_name_id)
+        return sorted(check_name_id_list)
+
+    def _control_sort_key(self, control: str) -> (str, int, int):
+        """Fabricate sort key."""
+        k1 = control.split('-')[0]
+        k2 = int(control.split('-')[1].split('.')[0])
+        if '.' in control:
+            k3 = int(control.split('-')[1].split('.')[1])
+        else:
+            k3 = 0
+        return (k1, k2, k3)
+
+    def _oscal_namify(self, control: str) -> str:
+        """Rectify parenthesized numbers in controls."""
+        control = control.replace('(', '.')
+        control = control.replace(')', '')
```

Comment:
Can simplify by chaining the two replace calls into one line.

Reply:
I dunno, for one less line my eye likes the code as is..